src/local-time.lisp

```diff
@@ -573,17 +573,16 @@ In other words:
             "timestamp-minimize-part called with invalid part ~a (expected one of ~a)"
             part
             timestamp-parts)
-    (multiple-value-bind (nsec sec min hour day month year day-of-week daylight-saving-time-p offset)
```

Comment:
If `encode-timestamp` returns an error with a `NIL` timezone, the answer is to fix `encode-timestamp` not ignore the offset parameter. Looking at the implementation of `encode-timestamp` it looks like `:timezone` is only used when `:offset` is `NIL`. Am I wrong?

Reply:
The old version says `:offset (if timezone nil offset)`, after getting `offset` from `(decode-timestamp timestamp :timezone timezone)`. The error would already happen in `decode-timezone`.

You could only get this error by calling (timezone-minimize-part timestamp part :timezone nil). There seem to be two styles of passing timezone:
   `&key (timezone *default-timezone*)`
and
  `&key timezone` combined with using it as `(or timezone *default-timezone*)`.
The latter is more robust, and I adopted it in several places, just in case. But there could also be a more strict checking that timezone is always a timezone in exported function.

Which style do you prefer?